src/plugin/plugin.js

```diff
@@ -2,5 +2,31 @@
  * This is the model class for a plugin.
  */
 class Plugin {
-	
-}
\ No newline at end of file
+	constructor({name, dependencies, author, path} = {}) {
+        this.name = name;
+        this.dependencies = dependencies;
+        this.author = author;
+        this.path = path;
+    }
+
+    addTrueDependencies(plugins) {
+        this.trueDependencies = this.addTrueDependencies(plugins);
+    }
```

Comment:
It looks like this code is looping indefinitely. Am I getting this wrong or is it meant to be something like 

```suggestion
    addTrueDependencies(plugins) {
        this.trueDependencies = this.getTrueDependencies(plugins);
    }
```

Reply:
Yes that was a typo. Fixed it, well spotted.